tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -921,163 +1030,169 @@ public void testDescribeAtMinIsrPartitions(String quorum) {
-                scala.collection.immutable.Map$.MODULE$.empty(), topicConfig
-        );
-        TestUtils.createTopicWithAdmin(adminClient, notUnderMinIsrTopic, scalaBrokers, scalaControllers, partitions, replicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
-        TestUtils.createTopicWithAdmin(adminClient, offlineTopic, scalaBrokers, scalaControllers, negativePartition, negativeReplicationFactor,
-                offlineReplicaAssignmentMap, new Properties()
-        );
-        TestUtils.createTopicWithAdmin(adminClient, fullyReplicatedTopic, scalaBrokers, scalaControllers, negativePartition, negativeReplicationFactor,
-                fullyReplicatedReplicaAssignmentMap, new Properties()
-        );
+    @ClusterTemplate("generate1")
+    public void testDescribeUnderMinIsrPartitionsMixed() throws InterruptedException {
+        try (Admin adminClient = clusterInstance.createAdminClient()) {
+            String underMinIsrTopic = "under-min-isr-topic";
+            String notUnderMinIsrTopic = "not-under-min-isr-topic";
+            String offlineTopic = "offline-topic";
+            String fullyReplicatedTopic = "fully-replicated-topic";
+            int partitions = 1;
+            short replicationFactor = 6;
+
+            List<NewTopic> newTopics = new ArrayList<>();
+
+            Map<Integer, List<Integer>> fullyReplicatedReplicaAssignmentMap = new HashMap<>();
+            fullyReplicatedReplicaAssignmentMap.put(0, Arrays.asList(1, 2, 3));
+
+            Map<Integer, List<Integer>> offlineReplicaAssignmentMap = new HashMap<>();
+            offlineReplicaAssignmentMap.put(0, Arrays.asList(0));
+
+            Map<String, String> topicConfig = new HashMap<>();
+            topicConfig.put(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, "6");
+
+            newTopics.add(new NewTopic(underMinIsrTopic, partitions, replicationFactor).configs(topicConfig));
+            newTopics.add(new NewTopic(notUnderMinIsrTopic, partitions, replicationFactor));
+            newTopics.add(new NewTopic(offlineTopic, offlineReplicaAssignmentMap));
+            newTopics.add(new NewTopic(fullyReplicatedTopic, fullyReplicatedReplicaAssignmentMap));
+
+            adminClient.createTopics(newTopics);
+            clusterInstance.waitForTopic(underMinIsrTopic, partitions);
+            clusterInstance.waitForTopic(notUnderMinIsrTopic, partitions);
+            clusterInstance.waitForTopic(offlineTopic, partitions);
+            clusterInstance.waitForTopic(fullyReplicatedTopic, partitions);
+
+            clusterInstance.shutdownBroker(0);
+            Assertions.assertEquals(5, clusterInstance.aliveBrokers().size());
+
+            TestUtils.waitForCondition(
+                    () -> clusterInstance.aliveBrokers().values().stream().allMatch(broker -> broker.metadataCache().getPartitionInfo(underMinIsrTopic, 0).get().isr().size() < 6 &&
```

Comment:
this line is too long can we break this into multiple lines. Maybe something like this 
```
TestUtils.waitForCondition(
                    () -> clusterInstance.aliveBrokers().values().stream()
                            .allMatch(broker ->
                                    broker.metadataCache().getPartitionInfo(underMinIsrTopic, 0).get().isr().size() < 6 &&
                                    broker.metadataCache().getPartitionInfo(offlineTopic, 0).get().leader() == MetadataResponse.NO_LEADER_ID
                            ),
                    60000, "Timeout waiting for partition metadata propagating to brokers for underMinIsrTopic topic"
            );
     ```